lib/retry.ex

```diff
@@ -146,6 +171,27 @@ defmodule Retry do
     end
   end
 
+  defmacro retry_while([with: stream_builder, acc: acc_initial], do: block),
+    do: do_retry_value([acc: acc_initial, with: stream_builder], do: block)
+
+  defmacro retry_while([acc: acc_initial, with: stream_builder], do: block),
+    do: do_retry_value([acc: acc_initial, with: stream_builder], do: block)
+
+  defp do_retry_value([acc: acc_initial, with: stream_builder], do: block) do
```

Comment:
I think this can be simplified to

```elixir
  defp do_retry_value([acc: acc_initial, with: stream_builder], do: block) do
    quote do
      unquote(delays_from(stream_builder))
      |> Enum.reduce_while(unquote(acc_initial), fn delay, acc ->
        :timer.sleep(delay)

        case acc do
          unquote(block)
        end
      end)
    end
  end

```